bus/src/main/java/org/hawkular/metrics/component/publish/InsertedDataSubscriber.java

```diff
@@ -64,8 +68,21 @@
 
     private Subscription subscription;
 
+    @Inject
+    private PublishCommandTable publish;
+
+    @Inject
+    @Configurable
+    @ConfigurationProperty(METRICS_PUBLISH_PERIOD)
+    private String publishPeriodProperty;
+    private int publishPeriod;
+
     public void onMetricsServiceReady(@Observes @ServiceReady ServiceReadyEvent event) {
-        Observable<List<Metric<?>>> events = event.getInsertedData().buffer(50, TimeUnit.MILLISECONDS, 100)
+        publishPeriod = getPublishPeriod();
+        Observable<List<Metric<?>>> events = event.getInsertedData()
+                .filter(m -> m.getType() != MetricType.STRING)
+                .filter(m -> publish.isPublished(m.getMetricId()))
```

Comment:
> It's unfortunate that Hmetrics decided to make metric uniqueness involve type. I think it may have been better if they assigned a type to a metric but still disallowed duplicate metricIds within a tenant. 

Metrics does not check or enforce uniqueness of ids. Performing those kinds of checks would have to be done on every write which would be bad performance-wise. There is another important reason why the type is part of the id. We use a statically typed schema
There are a couple reasons for having the type as part of the id. Values are stored either as a double, long, text, etc. We infer the data type of values based on the URI of the HTTP request. We need to store that type info somewhere. Storing it as non-primary key column doesn't make sense as it would have to be repeated for every data point. Alternatively, we could have used a separate table for each metric type, but we would still be in the same situation.